crates/services/shared-sequencer/src/lib.rs

```diff
@@ -0,0 +1,269 @@
+        // we calculate the actual amount and fee to use in real transaction.
+        let dummy_amount = Coin {
+            amount: 0,
+            denom: self.coin_denom.clone(),
+        };
+
+        let dummy_fee = Fee::from_amount_and_gas(dummy_amount, 0u64);
+
+        let dummy_payload = self
+            .make_payload(
+                timeout_height,
+                dummy_fee,
+                signer,
+                account,
+                order,
+                topic,
+                data.clone(),
+            )
+            .await?;
+
+        let used_gas = http_api::estimate_transaction(
+            &self.endpoints.blockchain_rest_api,
+            dummy_payload,
+        )
+        .await?;
+
+        let used_gas = used_gas.saturating_mul(2); // Add some buffer
+
+        let amount = Coin {
+            amount: self.gas_price.saturating_mul(used_gas as u128),
+            denom: self.coin_denom.clone(),
+        };
+
+        let fee = Fee::from_amount_and_gas(amount, used_gas);
+        let payload = self
+            .make_payload(timeout_height, fee, signer, account, order, topic, data)
+            .await?;
+
+        let r = self.tendermint()?.broadcast_tx_sync(payload).await?;
+        if r.code.is_err() {
+            return Err(PostBlobError { message: r.log }.into());
+        }
+        Ok(())
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    async fn make_payload<S: Signer>(
+        &self,
+        timeout_height: u32,
+        fee: Fee,
+        signer: &S,
+        account: AccountMetadata,
+        order: u64,
+        topic: [u8; 32],
+        data: Bytes,
+    ) -> anyhow::Result<Vec<u8>> {
+        let sender_account_id = self.sender_account_id(signer)?;
+
+        let msg = MsgPostBlob {
+            from: sender_account_id.to_string(),
+            order: order.to_string(),
+            topic: Bytes::from(topic.to_vec()),
+            data,
+        };
+        let any_msg = cosmrs::Any {
+            type_url: "/fuelsequencer.sequencing.v1.MsgPostBlob".to_owned(),
+            value: msg.encode_to_vec(),
+        };
+        let tx_body = tx::Body::new(vec![any_msg], "", timeout_height);
+
+        let sender_public_key = signer.public_key();
+        let signer_info =
+            SignerInfo::single_direct(Some(sender_public_key), account.sequence);
+        let auth_info = signer_info.auth_info(fee);
+        let sign_doc = SignDoc::new(
+            &tx_body,
+            &auth_info,
+            &self.ss_chain_id,
+            account.account_number,
+        )
+        .map_err(|err| anyhow!("{err:?}"))?;
+
+        let sign_doc_bytes = sign_doc
+            .clone()
+            .into_bytes()
+            .map_err(|err| anyhow!("{err:?}"))?;
+        let signature = signer.sign(&sign_doc_bytes).await?;
+
+        // Convert the signature to non-normalized form
+        let mut signature_bytes = *signature;
+        signature_bytes[32] &= 0x7f;
```

Comment:
A nit: It could be a little bit more clear if we have the `fn non_normalized()` added to `impl Signature`.

Reply:
Yep. This has to be done in  `fuel-vm`, though. I'll open an issue: https://github.com/FuelLabs/fuel-vm/issues/886